cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/monitor/task/LoadMonitorTaskRunner.java

```diff
@@ -262,21 +264,29 @@ public void shutdown() {
   /**
    * Pause the scheduled sampling tasks..
    */
-  public void pauseSampling() {
+  public synchronized void pauseSampling() {
     if (_state.get() != PAUSED && !_state.compareAndSet(RUNNING, PAUSED)) {
       throw new IllegalStateException("Cannot pause the load monitor because it is in " + _state.get() + " state.");
     }
+    _awaitExecution = true;
   }
 
   /**
    * Resume the scheduled sampling tasks.
    */
-  public void resumeSampling() {
+  public synchronized void resumeSampling() {
     if (_state.get() != RUNNING && !_state.compareAndSet(PAUSED, RUNNING)) {
       throw new IllegalStateException("Cannot resume the load monitor because it is in " + _state.get() + " state");
     }
+    _awaitExecution = false;
   }
 
+  /**
+   * Allow tasks to know if executor wishes to stop sampling b/c it wants to execute
+   */
+  public Boolean awaitingExecution() {
```

Comment:
`public Boolean awaitingExecution()` -> `public boolean awaitingPauseSampling()`

1. Can we use `boolean` in the response? -- i.e. `Boolean` is not needed, because the response could not be `null`. 
2. Can we also make the function name change (see my previous comment)?
3. Can we also update the JavaDoc to reflect the use of this new parameter?

Reply:
For item 3, what JavaDoc do we need to update?